discordpp/categories/guild.hh

```diff
@@ -492,3 +491,79 @@ sptr<const std::string> render_target() override {
+    NEW_FIELD(snowflake, guild_id, USEDBY(target))                             \
+    NEW_FIELD(std::string, query, USEDBY(target))                              \
+    NEW_FIELD(int, limit, USEDBY(target))                                      \
+    STATIC_FIELD(std::string, method, "GET")                                   \
+    HIDE_FIELD(target)                                                         \
+    HIDE_FIELD(type)                                                           \
+    HIDE_FIELD(body)                                                           \
+    FORWARD_FIELD(handleWrite, onWrite, )                                      \
+    FORWARD_FIELD(handleRead, onRead, )
+
+#include <discordpp/macros/defineCallOpen.hh>
+protected:
+sptr<const std::string> render_target() override {
+    std::string out = fmt::format("/guilds/{}/members/search", *_guild_id);
+    bool first = true;
+    if (!_guild_id) {
+        throw std::logic_error("Search Guild Members needs a Guild ID");
+    }
+    if(!_query) {
+        throw std::logic_error("Search Guild Members needs a Query");
+    } else {
+        out += fmt::format("{}query={}", first ? "?" : "&", *_query);
+        first = false;
+    }
+    if (_limit) {
+        out += fmt::format("{}limit={}", first ? "?" : "&", *_limit);
+        first = false;
+    }
+    return std::make_shared<const std::string>(out);
+}
+#include <discordpp/macros/defineCallClose.hh>
+
+//https://discord.com/developers/docs/resources/guild#modify-current-user-nick
+// Verified by Westlanderz/SenpaiR6#1717
+#define Bot PluginEndpoints
+#define Parent JsonCall
+#define Class ModifyCurrentUserNickCall
+#define function modifyCurrentUserNick
+#define Fields                                                                 \
+    NEW_FIELD(snowflake, guild_id, USEDBY(target))                             \
+    NEW_FIELD(std::string, nick, USEDBY(payload))                              \
+    STATIC_FIELD(std::string, method, "PATCH")                                 \
+    HIDE_FIELD(target)                                                         \
+    HIDE_FIELD(type)                                                           \
+    HIDE_FIELD(body)                                                           \
+    HIDE_FIELD(payload)                                                        \
+    FORWARD_FIELD(handleWrite, onWrite, )                                      \
+    FORWARD_FIELD(handleRead, onRead, )
+
+#include <discordpp/macros/defineCallOpen.hh>
+protected:
+sptr<const std::string> render_target() override {
+    if (!_guild_id) {
+        throw std::logic_error("Search Guild Members needs a Guild ID");
+    }
+    return std::make_shared<const std::string>(
+        fmt::format("/guilds/{}/members/@me/nick", *_guild_id));
+}
+sptr<const json> render_payload() override {
+    json out;
+    if(_nick) {
+        out["nick"] = *_nick;
+    } else {
+        out["nick"] = "";
```

Comment:
Could you test what happens when the field is empty vs. missing? If so it should be a `std::optional` with a test for that too.

Reply:
if its missing it does not do anything that is why i added the the empty quotes

Reply:
Even if the user has a nickname already? Weird. I wonder why they'd include that.